src/grep.rs

```diff
@@ -152,6 +157,7 @@ fn grep_offset(
     }
   };
 
+  let buffer = buffer.as_ref();
```

Comment:
Could you move this to the beginning of the function, just like in `grep_filename`? Just to keep things the same way.

Reply:
Sure.